DataCleaningScripts/new_moon_numbers.r

```diff
@@ -53,30 +55,48 @@ update_moon_dates = function() {
   # load rodent trapping data
   trappingdat=read.csv("../Rodents/Portal_rodent_trapping.csv")  
   trappingdat$CensusDate = as.Date(paste(trappingdat$Year,trappingdat$Month,trappingdat$Day,sep='-'))
-  # load new moon dates
-  newmoons=read.csv(url('https://www.dropbox.com/s/z9kq7qvu2r7uwzu/moon-phases-1977-2018-America_Phoenix.csv?dl=1'),header = TRUE, stringsAsFactors = FALSE) %>%
-    filter(phaseid==1) 
-  newmoons$date = as.Date(newmoons$date, format='%m/%d/%Y') 
-  #keep only newmoons past those currently assigned a newmoonnumber
-  newmoons=filter(newmoons,date>tail(moon_dates$NewMoonDate,n=1))
-  #add newmoonnumbers to newmoondates
-  newmoons$NewMoonNumber=tail(moon_dates$NewMoonNumber,n=1)+1:dim(newmoons)[1]
+  
   
   # proceed only if trappingdat has more recent dates than moon_dates
-  if (max(abs(trappingdat$Period),na.rm=T) > max(abs(moon_dates$Period),na.rm=TRUE)) {
-    
+  if (max(abs(trappingdat$Period),na.rm=T) > max(abs(moon_dates$Period),na.rm=TRUE)) {  
+   
+  #Get dates and period numbers of new data only  
     # extract trappingdat periods beyond those included in moon_dates
     newperiod = filter(trappingdat,abs(Period)>max(abs(moon_dates$Period),na.rm=TRUE))
     # reduce new trapping data to two columns: Period and CensusDate
-    newperiod_dates = find_first_trap_night(newperiod)
+    newperiod_dates = find_first_trap_night(newperiod)  
+     
+  # get new moon dates
+    #define date range for newmoon dates
+    first=newperiod_dates$CensusDate[1]-30; year=year(first); month=month(first) #shifting back a month just to not skip any
+    newmoondates=htmltab(doc=paste("http://aa.usno.navy.mil/cgi-bin/aa_phases.pl?year=",year,"&month=",month,"&day=1&nump=50&format=t", sep=""))
+    newmoondates=gsub('.{6}$', '', newmoondates$"Date and Time (Universal Time)"[newmoondates$"Moon Phase" == "New Moon"])
+    newmoondates = as.Date(ymd(newmoondates, format='%Y %m %d'))
+  
+  #Set up dataframe for new moon dates to be added
+  newmoons=data.frame(NewMoonNumber= NA, NewMoonDate = as.Date(newmoondates), Period = NA, CensusDate = as.Date(NA))
+    #keep only newmoon dates past those currently assigned a newmoonnumber
+    newmoons=filter(newmoons,NewMoonDate>tail(moon_dates$NewMoonDate,n=1))
+    #add newmoonnumbers to newmoondates
+    newmoons$NewMoonNumber=tail(moon_dates$NewMoonNumber,n=1)+1:dim(newmoons)[1]
+
+  #Match new census dates to moon dates
+
+    # match new period to closest NewMoonDate, 
+    for(i in 1:dim(newperiod_dates)[1]) {
+      closest = closest_newmoon(newperiod_dates$CensusDate[i],newmoondates)
+      newmoons$CensusDate[newmoons$NewMoonDate==newmoondates[closest]]=newperiod_dates$CensusDate[i]
+      newmoons$Period[newmoons$NewMoonDate==newmoondates[closest]]=newperiod_dates$Period[i]
+        }
```

Comment:
I like that idea
how bout we also change the closest_newmoon function so it returns the actual date instead of an index -- then "newmoondates[closest]" becomes just "closest"